src/symbolic_dimensions.jl

```diff
@@ -314,7 +318,19 @@ module SymbolicUnitsParse
 
     const UNIT_SYMBOLS_EXIST = Ref{Bool}(false)
     const UNIT_SYMBOLS_LOCK = Threads.SpinLock()
-    function _generate_unit_symbols()
+    function _generate_unit_symbols(; testing=Val(false))
```

Comment:
Just an FYI — the changes in this PR completely address what I observed in #51! I checked out this branch with my `AstrophysicalCalculations` package, and found the expected warning. Then I changed my code to use the `Quantity(..., SymbolicDimensions; pc=1)` constructor, as instructed by the warning, and my package precompiled as expected.

Reply:
Awesome!

The only reason I'm hesitating to merge is I'm not sure there are any cases where you may actually need to run through that function. I ran some testing and it seemed to hit that error even without necessarily precompiling a module with it... So maybe we could just leave the warning, but still allow it to execute as normal?

Reply:
>  I ran some testing and it seemed to hit that error even without necessarily precompiling a module with it...

That's interesting. I don't know enough about Julia's stages of compilation / precompilation. I think the warning is great; if I saw it before making #51, I would have used the `Quantity` constructor instead of filing an issue. Of course, no rush merging anything from my point of view; `AstrophysicalCalculations` is a "brush up on physics before graduate school" kind of package 😉 and the constructor is a great solution as-is.